### DISCUSSION TITLE
support of .vtp file format

### DISCUSSION THREAD

**Discussion by tusharsable** [iTwin/itwinjs-core#2386] [Q&A]
Can we open custom .vtp format file and place the mesh in an iModel opened in viewer.

**Comment by pmconne** [comment]
iModel.js frontend does not have direct access to the file system so you cannot open files directly. Your options include:
- Convert the mesh to a [supported 3D tile format](https://www.ogc.org/standards/3DTiles); or
- Import the mesh into an iModel as one or more geometric elements; or
- Implement and register with your viewport a [TiledGraphicsProvider](https://github.com/imodeljs/imodeljs/blob/92cd43ebbaa0e3753da882c6ce86dfbba5b64d32/core/frontend/src/tile/TiledGraphicsProvider.ts#L19) that can obtain the .vtp data and display it as meshes.
